File: python/report.py
from item import Item
from pricing import PricingRegistry


class Report:

    def __init__(self, items: [Item]):
        self.items = items
        self.pricing_registry = PricingRegistry()

    def __str__(self):
        header = f"{'name': <20}{'quality': >10}{'sell in': >10}{'price': >10}\n"
        table = "\n".join(
            f"{item.name: <20}" +
            f"{item.quality: >10}" +
            f"{item.sell_in: >10}" +
            f"{self.pricing_registry.get_price(item): >10.2f}"
            for item in self.items
        )
        return header + table

File: csharp/PimpedRose/Report.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace PimpedRose
{

    public class Report
    {
        List<Item> _items;
        PricingRegistry _pricingRegistry = new PricingRegistry();

        public Report(List<Item> items)
        {
            this._items = items;
        }


        public override string ToString()
        {
            var headerFormat = "{0,-20} {1,10} {2,10} {3,10}";
            var format = "{0,-20} {1,10} {2,10} {3,10:C}";
            var header = String.Format(
                headerFormat,
                "name",
                "quality",
                "sell in",
                "price"
            );
            var table = _items.Select(i=>String.Format(
                    format,
                    i.Name,
                    i.Quality,
                    i.SellIn,
                    _pricingRegistry.GetPrice(i)
                ));
            return header + "\n" + String.Join("\n", table);
        }

        public static Report createReport(List<Item> x)
        {
            return new Report(x);
        }
    }
}
